docs/workflow/testing/coreclr/running-aspnet-benchmarks-with-crossgen2.md

```diff
@@ -0,0 +1,231 @@
+described in the previous section).
+    * _Application_: Here we choose which job will be selected as the application
+    to run the benchmarks on, as well as other variables.
+    * _Load_: This is the tool that will generate and send the requests to
+    benchmark the web application. In this example, we are using `wrk` with a
+    warmup of 5 seconds and running the test for 60 seconds. In this example,
+    we are using the `json` headers for the load generation. There are various
+    headers that can be used, and these are defined in `wrk.yml`, which is
+    referenced at the top of this configuration file.
+
+* **Profiles**: The profiles describe the machines where the benchmarks will
+be run. This information was provided by the ASP&#46;NET team, who is in charge
+of these servers. In our example, there are two profiles, one for Windows,
+and one for Linux.
+
+## Run the Benchmarks
+
+Once you have your configuration file and CoreCLR built, it's time to run the
+initial benchmarks.
+
+### Initial Application
+
+From the `BenchmarksDriver2` folder, run the following command.
+
+On Windows:
+
+```powershell
+dotnet run -- --config crossgen2-benchmarks.yml --scenario json --profile aspnet-physical-win
+--application.options.fetch true
+```
+
+On Linux:
+
+```bash
+dotnet run -- --config crossgen2-benchmarks.yml --scenario json --profile aspnet-physical-lin
+--application.options.fetch true
+```
+
+Splitting and analyzing the previous command:
+
+* `--config crossgen2-benchmarks.yml`: This selects the configuration file to use.
+* `--scenario json`: This runs the scenario labelled as _json_ in the configuration file.
+* `--profile aspnet-physical-win`: This chooses the Windows profile from the configuration file.
+* `--application.options.fetch true`: This downloads the built application used for
+the benchmarks. We need these files to apply _crossgen2_ and then compare the benchmarks.
+Note that `application` is just the label given in the configuration file.
+
+At the end of the run, the tool will print a summary of statistics regarding
+how the performance went.
+
+### Crossgen2
+
+Grab the downloaded zip file with the application and extract it somewhere else.
+This is to avoid mixing up stuff or losing it when running `git clean` or the like.
+
+In this example, we will be using a new folder called `results` outside of the
+repos. Here, extract the zip into a folder we will refer to as `application`.
+Next, create another folder within `results` called `composite`. This is where
+the crossgen2'd assemblies will be stored.
+
+Now, go to your _Core\_Root_ inside the _runtime_ repo. From there, apply _crossgen2_
+using the following command.
+
+On Windows:
+
+```powershell
+CoreRun.exe \runtime\artifacts\bin\coreclr\Windows_NT.x64.Release\crossgen2\crossgen2.dll
+--Os --composite -o \path\to\results\composite\TotalComposite.dll \path\to\results\application\*.dll
```

Comment:
Adding @AntonLapounov since this will be simpler once composite support is added in the sdk. After that perhaps we can work with asp.net team to get that option added as part of the profile.